### PR TITLE
Gracefully reconnect to socket on error

### LINKED ISSUE #23: Handle socket disconnect/reconnect
In the event of docker socket disconnect, app need to gracefully handle disconnection / reconnection

### PR DESCRIPTION
**Pull request by thisisommore**
fixes #23

### CODE REVIEW

**Comment by thisisommore** [comment]
Example
Screenshot_20211103_224127

File: main.go [hunk 1]
```diff
@@ -101,18 +102,27 @@ func handleUfwRule(ch <-chan ufwEvent) {
...
 	cli, err := client.NewClientWithOpts(client.FromEnv)
 	if err != nil {
 		panic(err)
 	}
+	return cli
+}
 
+func addFilters(cli *client.Client, ctx *context.Context) (<-chan events.Message, <-chan error) {
+	_, cancelContext := context.WithCancel(*ctx)
+	cancelContext()
```
Comment (shinebayar-g) [inline main.go:115]: Could you tell me what it does? :)

File: main.go [hunk 2]
```diff
@@ -145,6 +155,10 @@ func main() {
 		case err := <-errors:
 			if err != nil {
 				fmt.Println("ufw-docker-automated: Received an error:", err)
+				time.Sleep(3 * time.Second)
+				fmt.Println("reconnecing")
```
Comment (shinebayar-g) [inline main.go:159]: Maybe?

```diff
-fmt.Println("reconnecing")
+ufw-docker-automated: reconnecting`
```

**Review by shinebayar-g — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (thisisommore) [reply main.go:115]: Cancels context, and therefore any go channels might be cleared I guess, let me know if I am wrong here, also I will remove it you think there is no need of this.

File: main.go (hunk 2 above)
Comment (thisisommore) [reply main.go:159]: Cool!

File: main.go (hunk 1 above)
Comment (shinebayar-g) [reply main.go:115]: Yeah, I'm not that familiar with Go contexts... For now best feedback I can give is as long as it works correct ... we'll keep it.

Comment (thisisommore) [reply main.go:115]: Agreed

**Review by shinebayar-g — APPROVED** [review]

**Pull request merged**